### PR TITLE
YAML definitions

### PR DESCRIPTION
**Pull request by jm9e**

### CODE REVIEW

File: slang.go [hunk 1]
```diff
@@ -111,7 +118,7 @@ func ParsePortReference(connStr string, par *core.Operator) (*core.Port, error)
 
 // readOperatorDef reads the operator definition for the given file and replaces all generic types according to the
 // generics map given. The generics map must not contain any further generic types.
-func readOperatorDef(opDefFilePath string, generics map[string]*core.PortDef, pathsRead []string) (core.OperatorDef, error) {
+func readOperatorDef(opDefFile string, generics map[string]*core.PortDef, pathsRead []string) (core.OperatorDef, error) {
```
Comment (td5r) [inline slang.go]: Wouldn't it be better, when `readOperaterDef` expects a filepath instead of file base name?

Comment (td5r) [reply slang.go]: everytime readOperatorDef calls itself, it specify the correct `opDefFile`Path

**Comment by td5r** [comment]
OperatorList is defined within core/def.go, which is nice, but why is MapStr defined within package utils?

**Comment by jm9e** [comment]
The idea is that all utility strucs and methods which are of general nature (like MapStr) are in package utils.

File: slang.go (hunk 1 above)
Comment (jm9e) [reply slang.go]: You are right. I wasn't happy with the solution either. I have changed it.

**Review by td5r — APPROVED** [review]

**Pull request merged**